### PR TITLE
Updated edit button focus

### PR DESCRIPTION
**Pull request by noahgelman**
## Description
Fixes issue noted in the following ticket: https://github.com/department-of-veterans-affairs/va.gov-team/issues/12338

## Original issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/12338#issuecomment-940113878

## Acceptance criteria
- [ ] Focus edit button when done editing

### CODE REVIEW

**Comment by AdamFreemer** [comment]
@noahgelman - It seems your changes are causing a failure on spec `src/applications/edu-benefits/tests/feedback-tool/01.feedback-tool-keyboard.cypress.spec.js` I confirmed by adding the code removed back in (please see my latest commit).

**Comment by noahgelman** [comment]
@AdamFreemer I see, ty

**Comment by AdamFreemer** [comment]
> @AdamFreemer I see, ty

@noahgelman -- just for clarity, are you working on a code change to this task that will pass tests?

File: src/platform/forms-system/src/js/utilities/ui/index.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ export function focusElement(selectorOrElement, options) {
   const el = $(selectorOrElement);
 
   if (el) {
+    // added back in below conditional for confirmation of broken test coverage
```
Comment (bkjohnson) [inline src/platform/forms-system/src/js/utilities/ui/index.js:13]: I'm just seeing a single comment - is there supposed to be more here?

**Comment by AdamFreemer** [comment]
@bkjohnson  -- I've been waiting for @noahgelman to respond with an update this PR by him, it's been a couple days with no response.

**Comment by noahgelman** [comment]
@AdamFreemer I am circling back around to this later today

File: src/platform/forms-system/src/js/utilities/ui/index.js [hunk 2]
```diff
@@ -10,12 +10,6 @@ export function focusElement(selectorOrElement, options) {
   const el = $(selectorOrElement);
 
   if (el) {
-    if (el.tabIndex === 0) {
-      el.setAttribute('tabindex', '0');
-    }
-    if (el.tabIndex < 0) {
-      el.setAttribute('tabindex', '-1');
-    }
     el.focus(options);
```
Comment (bkjohnson) [inline src/platform/forms-system/src/js/utilities/ui/index.js:13]: Could we simplify this with optional chaining?

```suggestion
  el?.focus(options);
```

**Comment by noahgelman** [comment]
We're looking into some build issues that may be causing the build errors.
Currently investigating

File: src/platform/forms-system/src/js/utilities/ui/index.js (hunk 2 above)
Comment (noahgelman) [reply src/platform/forms-system/src/js/utilities/ui/index.js:13]: Yeah, but I'm old, and ES2020 is too young and cool for me

**Review by bethdeconinck — APPROVED** [review]
Tested in staging and confirmed expected focus behavior (edit button received focus after making changes, focus visible focus indicator present).

File: src/platform/forms-system/src/js/utilities/ui/index.js (hunk 2 above)
Comment (bkjohnson) [reply src/platform/forms-system/src/js/utilities/ui/index.js:13]: We don't have an ESLint rule for it, but optional chaining should be preferred when possible.

**Review by bkjohnson — CHANGES_REQUESTED** [review]
The issue is written specifically about the review page, but this `focusElement` helper is used all over the place in other components. Do we know that removing the `tabindex` setters won't cause any regressions? Here's an example of something that might no longer work:

https://github.com/department-of-veterans-affairs/vets-website/blob/4f6e8739bb8fdcf5dc21e39b3e1e208e4b3aefff/src/applications/personalization/profile/sass/profile.scss#L11-L13

The "Changes requested" aspect of this is specifically for the optional chaining - I'm more curious about the possible implications of removing this functionality at the platform level rather than in a more isolated component/helper.

---

**Edit**:

It looks like the `tabindex` setters may be the way they are [in order to provide IE11 support](https://github.com/department-of-veterans-affairs/vets-website/pull/10237/files#r293086906). Has this change been tested in IE11?

**Comment by timwright12** [comment]
@bkjohnson @k80bowman Should this be closed out or should the work be picked up?

**Comment by k80bowman** [comment]
@timwright12 closing

**Pull request closed without merging**